src/Uno.UI/UI/Xaml/DependencyObjectStore.cs

```diff
@@ -1123,22 +1123,32 @@ internal void UpdateResourceBindings(bool isThemeChangedUpdate, ResourceDictiona
 
 			foreach (var kvp in bindings)
 			{
-				var wasSet = false;
-				foreach (var dict in dictionariesInScope)
+				try
 				{
-					if (dict.TryGetValue(kvp.Value.ResourceKey, out var value, shouldCheckSystem: false))
+					var wasSet = false;
+					foreach (var dict in dictionariesInScope)
 					{
-						wasSet = true;
-						SetValue(kvp.Key, BindingPropertyHelper.Convert(() => kvp.Key.Type, value), kvp.Value.Precedence);
-						break;
+						if (dict.TryGetValue(kvp.Value.ResourceKey, out var value, shouldCheckSystem: false))
+						{
+							wasSet = true;
+							SetValue(kvp.Key, BindingPropertyHelper.Convert(() => kvp.Key.Type, value), kvp.Value.Precedence);
+							break;
+						}
 					}
-				}
 
-				if (!wasSet && isThemeChangedUpdate && kvp.Value.IsThemeResourceExtension)
+					if (!wasSet && isThemeChangedUpdate && kvp.Value.IsThemeResourceExtension)
+					{
+						if (ResourceResolver.TryTopLevelRetrieval(kvp.Value.ResourceKey, kvp.Value.ParseContext, out var value))
+						{
+							SetValue(kvp.Key, BindingPropertyHelper.Convert(() => kvp.Key.Type, value), kvp.Value.Precedence);
+						}
+					}
+				}
+				catch (Exception e)
```

Comment:
Maybe you should catch a more specific exception type here?

Reply:
Indeed. What kind of exception is raised ?

Reply:
An `ObjectDisposedException` was what I saw... but I don't really see why it should catch a specific exception type. We want to update the remaining bindings, regardless of if one of the previous ones fails for any reason. Otherwise part of the visual tree may fail to update.